joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/types/TypeRenderer.scala

```diff
@@ -55,7 +57,20 @@ object TypeRenderer {
           val fqName     = DescriptorUtils.getFqName(descriptor)
           val mappedType = JavaToKotlinClassMap.INSTANCE.mapKotlinToJava(fqName)
           if (mappedType != null) {
-            stripped(renderer.renderFqName(mappedType.asSingleFqName().toUnsafe))
+            val fqName                   = mappedType.asSingleFqName()
+            val nonUnwrappedRender       = stripped(renderer.renderFqName(fqName.toUnsafe))
+            val isWrapperOfPrimitiveType = JvmPrimitiveType.isWrapperClassName(fqName)
+
+            if (unwrapPrimitives && isWrapperOfPrimitiveType) {
+              JvmPrimitiveType
+                .values()
+                .toList
+                .filter(_.getWrapperFqName.toString == fqName.toString)
+                .map(_.getJavaKeywordName)
```

Comment:
Could be also done with collect. Just a matter of style.

Reply:
keeping it as it is for now, will tackle on a cleanup round of the whole frontend